src/plugins/saved_objects_management/public/management_section/objects_table/components/flyout.tsx

```diff
@@ -841,11 +843,16 @@ export class Flyout extends Component<FlyoutProps, FlyoutState> {
   }
 
   renderFooter() {
-    const { isLegacyFile, status } = this.state;
+    const { isLegacyFile, status, selectedDataSourceId } = this.state;
     const { done, close } = this.props;
 
     let confirmButton;
 
+    let importButtonDisabled = false;
+    if (this.props.dataSourceEnabled && this.props.hideLocalCluster && !selectedDataSourceId) {
+      importButtonDisabled = true;
```

Comment:
Thanks for adding the check! I think we can simplify this to disabled = {this.props.dataSourceEnabled && this.props.hideLocalCluster && !selectedDataSourceId}. We can also extract the 3 checks as a function with a name that is clear what we are testing

Reply:
That's a good point, thanks for the comment.

Reply:
Hey, update on this comment.
I try to extract this logic as a function and put it into dataSourceManagement plugin. 
It is able to be consumed in other plugins by importing that function. 

But there is a problem.
When we try to import that extracted function into a different plugin, we need to do following staff to skip es-lint check:

```
// eslint-disable-next-line @osd/eslint/no-restricted-paths
import { isValidDataSourceId } from '../../../../data_source_management/public/components/utils';
```

And I have an offline sync up with @zengyan-amazon, skipping es-lint check is not a good idea and we should avoid it as much as we can.

The recommended option to call a component under a plugin is: 

1. Adding logic under a plugin A
2. Add plugin A as optional / required plugin of other plugins
3. Consume it lifecycle function

Since this judgement logic is not necessary to be centralized, it will be complex to follow the above way. So will keep it as it is.

Thanks for the comment~